src/granite/querying.cr

```diff
@@ -109,11 +136,23 @@ module Granite::Querying
     @@adapter.open { |db| db.exec(clause) }
   end
 
-  def query(clause = "", params = [] of DB::Any, &block)
+  def query(clause = "", params = [] of Granite::Fields::Type, &block)
     @@adapter.open { |db| yield db.query(clause, params) }
   end
 
   def scalar(clause = "", &block)
     @@adapter.open { |db| yield db.scalar(clause) }
   end
+
+  private def exec_exists(clause : String, params : Array(Granite::Fields::Type)) : Bool
+    @@adapter.exists? quoted_table_name, clause, params
+  end
+
+  private def build_find_by_clause(criteria : Hash(Symbol | String, Granite::Fields::Type))
+    criteria = criteria.dup
+    {
+      criteria.keys.map { |name| "#{quoted_table_name}.#{quote(name.to_s)} #{(criteria[name].nil? ? (criteria.delete(name); "IS NULL") : "= ?")}" }.join(" AND "),
+      criteria.values,
+    }
+  end
```

Comment:
I'm a short line, quick to read kind of guy. What do you think of writing this method as:

```crystal
private def build_find_by_clause(criteria : Hash(Symbol | String, Granite::Fields::Type))
  names = criteria.keys
  values = criteria.dup

  clauses = names.map do |name|
    if values[name]
      matcher = "= ?"
    else
      matcher = "IS NULL"
      values.delete name
    end

    "#{quoted_table_name}.#{quote name.to_s} #{matcher}"
  end

  { clauses.join(" AND "), values }
end
```